### PR TITLE
docs: add quickstart guide for k3s

### LINKED ISSUE #453: Quickstart with k3s

### PR DESCRIPTION
**Pull request by mkimuram**
Closes: #453

### CODE REVIEW

**Comment by mkimuram** [comment]
@mangelajo 

This might need to wait for 0.9 release, because 
- k3s requires network plugin detection fix (already merged in devel branch),
- Wait for change in `--natt=false` options from `--disable-nat`

However, making PR to get feedback early.

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+date: 2021-02-24T13:36:18+01:00
+title: "k3s"
+weight: 10
```
Comment (dfarrell07) [inline src/content/getting-started/quickstart/k3s/_index.md]: Let's make this weight 15 (or 20 and bump the current 20/30 by 10) so it comes after the KIND sandbox/CI environment in the left nav bar list. I think that's our recommended "default path" for folks.

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+---
+date: 2021-02-24T13:36:18+01:00
+title: "k3s"
```
Comment (dfarrell07) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
title: "K3s"
```

It seems like it's generally capitalized in the project's docs? https://github.com/k3s-io/k3s

(applies elsewhere)

**Review by dfarrell07 — CHANGES_REQUESTED** [review]
Thanks @mkimuram! A few minor suggestions inline. Otherwise it looks good from a docs prospective, although I haven't actually run through the steps.

**Comment by nyechiel** [comment]
@mkimuram can you please address the review comments? Let's get it merged :)

**Comment by mkimuram** [comment]
@nyechiel 

Thank you for your reminder. 

@dfarrell07

Thank you for your review and sorry for late response.
Fixed. PTAL

**Review by dfarrell07 — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
+---
+date: 2021-02-24T13:36:18+01:00
+title: "K3s"
+weight: 15
+---
+
+This quickstart guide covers the necessary steps to deploy two [K3s](https://k3s.io/) cluslters
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
This quickstart guide covers the necessary steps to deploy two [K3s](https://k3s.io/) clusters
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+guide focuses on Submariner deployment on clusters with non-overlapping Pod and Service CIDRs.
+For connecting clusters with overlapping CIDRs, please refer to the
+[Globalnet guide](../../architecture/globalnet/).
+
+### Prerequisites
+
+1. Prepare two nodes with
+[K3s-supported OS](https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#operating-systems)
+installed (These two nodes are referred to as node-a and node-b).
+2. Decide the Pod CIDR and the Service CIDR for each K3s cluster.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
2. Choose the Pod CIDR and the Service CIDR for each K3s cluster.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+### Deploy K3s Clusters
+
+#### Deploy cluster-a on node-a
+
+In this step you will deploy **cluster-a** on node-a using the the following IP ranges:
+
+| Pod CIDR     | Service CIDR |
+|--------------|--------------|
+|10.44.0.0/16  |10.45.0.0/16  |
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]: In  the Prerequisites, we instruct the user to choose the CIDRs but here we're telling them to use specific CIDRs.

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+1. Prepare two nodes with
+[K3s-supported OS](https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#operating-systems)
+installed (These two nodes are referred to as node-a and node-b).
+2. Decide the Pod CIDR and the Service CIDR for each K3s cluster.
+
+### Deploy K3s Clusters
+
+#### Deploy cluster-a on node-a
+
+In this step you will deploy **cluster-a** on node-a using the the following IP ranges:
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
In this step you will deploy **cluster-a** on node-a using the the following CIDR:
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+| Pod CIDR     | Service CIDR |
+|--------------|--------------|
+|10.44.0.0/16  |10.45.0.0/16  |
+
+To deploy K3s with the IP ranges, run below commands on node-a.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
To deploy K3s with the CIDRs, run the following commands on node-a.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+SERVICE_CIDR=10.45.0.0/16
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--cluster-cidr $POD_CIDR --service-cidr $SERVICE_CIDR" sh -s -
+```
+
+{{% notice tip %}}
+For more information on interacting with K3s, please refer to the [k3s documentation](https://rancher.com/docs/k3s/latest/en/quick-start/).
+{{% /notice %}}
+
+The kubeconfig file needs to be modified like below to make k8s API endpoint point to public IP of the node-a.
+If the IP should be different from the first field of `hostname -I`, modify the value manually.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
The kubeconfig file needs to be modified to set the Kubernetes API endpoint point to the public IP the node-a,
which is obtained from the first field yielded by `hostname -I`.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+```bash
+cp /etc/rancher/k3s/k3s.yaml kubeconfig.cluster-a
+IP=$(hostname -I | awk '{print $1}')
+sed -i 's/127.0.0.1/'$IP'/' kubeconfig.cluster-a
+```
+
+#### Deploy cluster-b on node-b
+
+In this step you will deploy **cluster-b** on node-b using the the following IP ranges:
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
In this step you will deploy **cluster-b** on node-b using the the following CIDRs:
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+| Pod CIDR     | Service CIDR |
+|--------------|--------------|
+|10.144.0.0/16 |10.145.0.0/16 |
+
+To deploy K3s with the IP ranges, run below commands on node-b.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
To deploy K3s with the CIDRs, run the following commands on node-b.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+```bash
+POD_CIDR=10.144.0.0/16
+SERVICE_CIDR=10.145.0.0/16
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--cluster-cidr $POD_CIDR --service-cidr $SERVICE_CIDR" sh -s -
+```
+
+The kubeconfig file needs to be modified like below to make k8s API endpoint point to public IP of the node-b.
+If the IP should be different from the first field of `hostname -I`, modify the value manually.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
The kubeconfig file needs to be modified to set the Kubernetes API endpoint point to the public IP the node-a,
which is obtained from the first field yielded by `hostname -I`.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+```bash
+cp /etc/rancher/k3s/k3s.yaml kubeconfig.cluster-b
+IP=$(hostname -I | awk '{print $1}')
+sed -i 's/127.0.0.1/'$IP'/' kubeconfig.cluster-b
+```
+
+Then, copy kubeconfig.cluster-b to node-a.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
Next, copy kubeconfig.cluster-b to node-a.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+
+#### Install `subctl` on node-a
+
+{{< subctl-install >}}
+
+#### Use cluster-a as Broker
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
#### Use cluster-a as the Broker
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+
+On node-a, run the below command.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
On node-a, run:
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+
+```bash
+subctl join --kubeconfig kubeconfig.cluster-a broker-info.subm --clusterid cluster-a --natt=false
+```
+
+##### Join node-b
+
+```bash
+subctl join --kubeconfig kubeconfig.cluster-b broker-info.subm --clusterid cluster-b --natt=false
+```
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]: I think we can reduce these sections.

#### Join cluster-a to the Broker

```bash
subctl join --kubeconfig kubeconfig.cluster-a broker-info.subm --clusterid cluster-a --natt=false
```

#### Join cluster-b to the Broker

```bash
subctl join --kubeconfig kubeconfig.cluster-b broker-info.subm --clusterid cluster-b --natt=false
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+
+You now have a Submariner environment that you can experiment with.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]: I don't think we need this statement.

```suggestion
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by mkimuram** [comment]
@tpantelis 

Thank you for your review. Fixed as suggested. 

> In the Prerequisites, we instruct the user to choose the CIDRs but here we're telling them to use specific CIDRs.

The third commit addresses this. Please check if it is OK.

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+    |-----------|--------------|--------------|
+    | cluster-a |10.44.0.0/16  |10.45.0.0/16  |
+    | cluster-b |10.144.0.0/16 |10.145.0.0/16 |
+
+### Deploy K3s Clusters
+
+#### Deploy cluster-a on node-a
+
+In this step you will deploy **cluster-a** on node-a using the the specific CIDRs.
+To deploy K3s with the CIDRs, run the following commands on node-a.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
In this step you will deploy K3s on node-a using the the **cluster-a** CIDRs.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+POD_CIDR=10.44.0.0/16
+SERVICE_CIDR=10.45.0.0/16
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--cluster-cidr $POD_CIDR --service-cidr $SERVICE_CIDR" sh -s -
+```
+
+{{% notice tip %}}
+For more information on interacting with K3s, please refer to the [k3s documentation](https://rancher.com/docs/k3s/latest/en/quick-start/).
+{{% /notice %}}
+
+The kubeconfig file needs to be modified to set the Kubernetes API endpoint to the public IP of the node-a,
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
The kubeconfig file needs to be modified to set the Kubernetes API endpoint to the public IP of node-a,
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+```bash
+cp /etc/rancher/k3s/k3s.yaml kubeconfig.cluster-a
+IP=$(hostname -I | awk '{print $1}')
+sed -i 's/127.0.0.1/'$IP'/' kubeconfig.cluster-a
+```
+
+#### Deploy cluster-b on node-b
+
+In this step you will deploy **cluster-b** on node-b using the the specific CIDRs.
+To deploy K3s with the CIDRs, run the following commands on node-b.
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
In this step you will deploy K3s on node-b using the the **cluster-b** CIDRs.
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+
+```bash
+POD_CIDR=10.144.0.0/16
+SERVICE_CIDR=10.145.0.0/16
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--cluster-cidr $POD_CIDR --service-cidr $SERVICE_CIDR" sh -s -
+```
+
+The kubeconfig file needs to be modified to set the Kubernetes API endpoint to the public IP of the node-b,
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
The kubeconfig file needs to be modified to set the Kubernetes API endpoint to the public IP of node-b,
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+
+On node-a, run:
+
+```bash
+subctl deploy-broker --kubeconfig kubeconfig.cluster-a --service-discovery
+```
+
+#### Join cluster-a and cluster-b to the Broker
+
+Join cluster-a to the Broker
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]: I think it's simpler to just make this its own section - avoids repeating "Join cluster-a ...".

```suggestion
#### Join cluster-a to the Broker
```

File: src/content/getting-started/quickstart/k3s/_index.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+
+```bash
+subctl join --kubeconfig kubeconfig.cluster-a broker-info.subm --clusterid cluster-a --natt=false
+```
+
+Join cluster-b to the Broker
```
Comment (tpantelis) [inline src/content/getting-started/quickstart/k3s/_index.md]:
```suggestion
#### Join cluster-b to the Broker
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by mkimuram** [comment]
@tpantelis 

Thank you for your review. Fixed and squashed the commit.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**